packages/next/src/build/swc/options.ts

```diff
@@ -217,7 +219,7 @@ function getBaseSWCOptions({
       serverComponents && !jest
         ? {
             isReactServerLayer,
-            dynamicIoEnabled: isDynamicIo,
+            dynamicIoEnabled: isDynamicIo || isCacheComponents,
```

Comment:
the rust type (`ReactServerComponents`) says it requires both, so that's probably why serde is yelling
```suggestion
            dynamicIoEnabled: isDynamicIo || isCacheComponents,
            cacheComponentsEnabled: isDynamicIo || isCacheComponents,
```